libs/protos/protos/sp_pipeline.proto

```diff
@@ -45,15 +45,51 @@ enum AbortCondition {
 // Conditions define how the SDK should handle a Wasm response in a step.
 // Should it continue executing the pipeline, should it abort, should it notify
 // and on_error.
+// TODO: de-pluralize this name
 message PipelineStepConditions {
   // Should we abort execution?
   AbortCondition abort = 1;
 
-  // Should we trigger a notification?
-  bool notify = 2;
+  bool notify = 2 [deprecated=true];
 
   // Should we include additional metadata that SDK should pass back to user?
   map<string, string> metadata = 3;
+
+  PipelineStepNotification notification = 4;
+}
+
+
+message PipelineStepNotification {
+  enum PayloadType {
+    // Same functionality as PAYLOAD_TYPE_EXCLUDE
+    PAYLOAD_TYPE_UNSET = 0;
+
+    // Default. No payload data included in notification
+    PAYLOAD_TYPE_EXCLUDE = 1;
+
+    // Entire payload content included in notification
+    PAYLOAD_TYPE_FULL_PAYLOAD = 2;
+
+    // Only specified paths of payload content included in notification
+    // Only works on JSON. Plaintext payloads will be ignored.
+    PAYLOAD_TYPE_SELECT_PATHS = 3;
+  }
+
+  // The UUIDs of the notification config to use
+  // This is kept separate to avoid having to configure slack/email settings
+  // every time and also because that config info is sensitive and is encrypted
+  repeated string notification_config_ids = 1;
+
+  PayloadType payload_type = 2;
+
+  // If type == paths, then we will look here for a list of json paths to include
+  // in the notification payload.
+  repeated string paths = 3;
+
+  // Notification configs for this pipeline. Only filled out in external API responses
```

Comment:
Not following the intent of this - remove if it's not needed / keep if you're still thinking it over or we can discuss in next standup.

Reply:
It's deprecated, it was previously used for including the actual notification configuration, such as slack config or email config, in the response for frontend